### ISSUE TITLE
Consider switching from HTTP to HTTPS for Iglu Central schemas

### ISSUE THREAD

**Issue by dnedev** [snowplow/iglu-central#1067]
Hey all,

We recently started using a new JSON schema validation framework for some events we emit internally to our Snowplow collector. However, the most recent version of the framework requires custom JSON schemas to retrieved from hosts over HTTPS.

My questions are:
- Would you consider switching/upgrading the Iglu Central deployment to start service schemas from both HTTP (for legacy use) and HTTPS?
- If so, would you consider updating the schema references use HTTPS, e.g. `http://iglucentral.com/schemas/com.snowplowanalytics.self-desc/schema/jsonschema/1-0-0#` -> `https://iglucentral.com/schemas/com.snowplowanalytics.self-desc/schema/jsonschema/1-0-0`

This would be of great help to us. The alternative for us would be to re-upload all schemas to our own Iglu repo (which only serves HTTPS requests), as well are replacing all schema reference for all custom event and context schemas.

Thanks,
Dimitar

**Comment by chuwy** [comment]
Hey @dnedev!

Thanks for raising this, it actually can become a very important thing for Iglu ecosystem.

> We recently started using a new JSON schema validation framework 

Is this [json-schema-validator](https://github.com/networknt/json-schema-validator)? We have ran into a similar problem with it because Iglu Scala Client (a core validation software for whole Snowplow ecosystem) also uses this library. We ended up [stuck on 1.0.39](https://github.com/snowplow/iglu-scala-client/issues/136), which still allows HTTP. I still don't know why this decision has been made by JSV maintainers, but to me it seems that it must be at least configurable, because of its breaking nature.

On the actual proposal - I agree we should do this eventually (just by means of getting rid of HTTP in general), but it is unfortunately not that easy to do. Mostly because our software (and JSV) treats the `http://iglucentral.com/schemas/com.snowplowanalytics.self-desc/schema/jsonschema/1-0-0#` as hard-coded identifier, which tells that attached schema is self-describing. Which means at least we need release a new Iglu Client that also adds support of HTTPS-based schema. But even after that, we cannot replace all URIs with HTTPS version on Iglu Central because there will remain a lot of legacy pipelines that use older Iglu Client. As a result, we probably would have to stick with migration strategy with Iglu Central mirror.

TL;DR - If we decide follow this path - it will likely take a lot of time.

As a more viable workaround, I'm planning to contact JSV maitainers and find out why this decision has been made and hopefully contribute a configuration flag that disables it. Or were you talking about some other library?

**Comment by chuwy** [comment]
Oh, just noticed you [raised this already](https://github.com/networknt/json-schema-validator/issues/314). Thanks! Joining the conversation.

**Comment by hisener** [comment]
Hey @chuwy, I filed https://github.com/networknt/json-schema-validator/pull/400 to introduce a configuration flag. PTAL if you have. 🙏

**Comment by chuwy** [comment]
Hey @hisener! Thank you very much! It unblocks us with Iglu Scala Client, but I think we still need to move to HTTPS at some point.
